### PR TITLE
fix: delete chart or dashboards with assoc reports

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Charts, databases and dashboards were not checking for associated `ReportSchedule`'s on delete, This PR fixes that for delete and bulk delete

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/commands/exceptions.py [hunk 1]
```diff
@@ -78,13 +78,21 @@ class ChartDeleteFailedError(DeleteFailedError):
     message = _("Chart could not be deleted.")
 
 
+class ChartDeleteFailedReportsExistError(ChartDeleteFailedError):
+    message = _("There are associated alerts or reports associated")
```
Comment (villebro) [inline superset/charts/commands/exceptions.py:82]: The word "associated" is used twice here. Would "The chart has associated alerts or reports" make sense?

File: superset/charts/commands/exceptions.py [hunk 2]
```diff
@@ -78,13 +78,21 @@ class ChartDeleteFailedError(DeleteFailedError):
...
     message = _("Changing this chart is forbidden")
 
 
-class ChartBulkDeleteFailedError(CreateFailedError):
+class ChartBulkDeleteFailedError(DeleteFailedError):
     message = _("Charts could not be deleted.")
 
 
+class ChartBulkDeleteFailedReportsExistError(ChartBulkDeleteFailedError):
+    message = _("There are associated alerts or reports associated")
```
Comment (villebro) [inline superset/charts/commands/exceptions.py:94]: Same here

File: superset/dashboards/commands/exceptions.py [hunk 3]
```diff
@@ -53,6 +53,10 @@ class DashboardBulkDeleteFailedError(CreateFailedError):
     message = _("Dashboards could not be deleted.")
 
 
+class DashboardBulkDeleteFailedReportsExistError(DashboardBulkDeleteFailedError):
+    message = _("There are associated alerts or reports associated")
```
Comment (villebro) [inline superset/dashboards/commands/exceptions.py:57]: Same here

File: superset/dashboards/commands/exceptions.py [hunk 4]
```diff
@@ -61,6 +65,10 @@ class DashboardDeleteFailedError(DeleteFailedError):
     message = _("Dashboard could not be deleted.")
 
 
+class DashboardDeleteFailedReportsExistError(DashboardDeleteFailedError):
+    message = _("There are associated alerts or reports associated")
```
Comment (villebro) [inline superset/dashboards/commands/exceptions.py:69]: Same here

File: superset/databases/commands/exceptions.py [hunk 5]
```diff
@@ -113,6 +113,10 @@ class DatabaseDeleteFailedError(DeleteFailedError):
     message = _("Database could not be deleted.")
 
 
+class DatabaseDeleteFailedReportsExistError(DatabaseDeleteFailedError):
+    message = _("There are associated alerts or reports associated")
```
Comment (villebro) [inline superset/databases/commands/exceptions.py:117]: same here

**Review by villebro — APPROVED** [review]
A minor wording nit, otherwise LGTM.

File: superset/charts/commands/exceptions.py (hunk 1 above)
Comment (dpgaspar) [reply superset/charts/commands/exceptions.py]: fixed

File: superset/charts/commands/exceptions.py (hunk 2 above)
Comment (dpgaspar) [reply superset/charts/commands/exceptions.py]: fixed

File: superset/dashboards/commands/exceptions.py (hunk 4 above)
Comment (dpgaspar) [reply superset/dashboards/commands/exceptions.py]: fixed

File: superset/databases/commands/exceptions.py (hunk 5 above)
Comment (dpgaspar) [reply superset/databases/commands/exceptions.py]: fixed

**Pull request merged**